kernel/async/monitor-common/async/Tasklet.hh

```diff
@@ -55,7 +55,7 @@ public:
 
   static constexpr size_t PAYLOAD_SIZE = CLSIZE - sizeof(TaskletBase) - sizeof(FunPtr);
 
-  Tasklet() : handler(nullptr) {}
+  Tasklet() : handler(nullptr) { TaskletBase::nextTasklet = UNUSED; }
```

Comment:
let's replace the constructor by initialisation values at the member variable's declaration.
We should add a check in the constructor:
`    ~Tasklet() { assert(isUnused()); }`